src/eos-metrics-instrumentation.c

```diff
@@ -284,6 +300,72 @@ record_live_boot (gpointer unused)
     return G_SOURCE_REMOVE;
 }
 
+static gchar *
+get_image_version_for_path (const gchar *path)
+{
+    ssize_t xattr_size;
+    gchar *image_version = NULL;
+
+    xattr_size = getxattr (path, EOS_IMAGE_VERSION_XATTR, NULL, 0);
+
+    if (xattr_size < 0)
+      return NULL;
+
+    image_version = g_malloc0 (xattr_size + 1);
+
+    xattr_size = getxattr (path, EOS_IMAGE_VERSION_XATTR,
+                           image_version, xattr_size);
+
+    /* this check is just in case the xattr has changed in between the
+     * size checks */
+    if (xattr_size < 0)
+      {
+        g_warning ("Error when getting 'eos-image-version' from %s: %s", path,
+                   strerror (errno));
+        g_free (image_version);
```

Comment:
I think it would be clearer to use `g_autofree gchar *image_version = NULL` and `return g_steal_pointer (&image_version);` on success; but this works.